### ISSUE TITLE
need apostrophe support

### ISSUE THREAD

**Issue by guvanch07** [slang-i18n/slang#177]
**Motivation**
when use the apostrophes in words (ex: Let's, dont't) slang generate \ backslash which is displayed on the UI with text

**Developer Experience**

**Comment by Tienisto** [comment]
Can you show your slang config?

**Comment by guvanch07** [comment]
`global_options:
  freezed:
    runs_before:
      - json_serializable
  json_serializable:
    runs_before:
      - retrofit_generator
targets:
  $default:
    builders:
      slang_build_runner:
        options:
          base_locale: en
          fallback_strategy: base_locale
          input_directory: assets/langs
          input_file_pattern: .json
          output_directory: lib/generated
          # output_file_pattern: .g.dart # deprecated, use output_file_name
          output_file_name: translations.g.dart
          output_format: single_file
          locale_handling: true
          namespaces: true
          translate_var: t
          enum_name: AppLocale
          translation_class_visibility: private
          # key_case: snake
          key_case: camel
          # key_map_case: camel
          # param_case: pascal
          param_case: camel
          # string_interpolation: double_braces
          string_interpolation: braces
          # TODO: change to false
          flat_map: true
          # TODO: comment
          translation_overrides: true
          timestamp: true
          maps:
            - models.dashboardPeriod.name
            - models.dashboardPeriod.versus
            - models.currency.fullName
            - models.lengthUnit.name
            - models.weightUnit.name
            - models.orderStatus.name
            - models.paymentStatus.name
            - models.productStatus.name
            - models.paymentMethodType.name
            - models.paymentMethodType.description
            - models.storeStatus.name
            - models.promoType.name
          pluralization:
            auto: cardinal
            cardinal:
              - someKey.apple
            ordinal:
              - someKey.place
          contexts:
            gender_context:
              enum:
                - male
                - female
              auto: false
              paths:
                - my.path.to.greet
          interfaces:
            PageData: onboarding.pages.*
            PageData2:
              paths:
                - my.path
                - cool.pages.*
              attributes:
                - String title
                - String? content`

**Comment by guvanch07** [comment]
```
global_options:
  freezed:
    runs_before:
      - json_serializable
  json_serializable:
    runs_before:
      - retrofit_generator
targets:
  $default:
    builders:
      slang_build_runner:
        options:
          base_locale: en
          fallback_strategy: base_locale
          input_directory: assets/langs
          input_file_pattern: .json
          output_directory: lib/generated
          # output_file_pattern: .g.dart # deprecated, use output_file_name
          output_file_name: translations.g.dart
          output_format: single_file
          locale_handling: true
          namespaces: true
          translate_var: t
          enum_name: AppLocale
          translation_class_visibility: private
          # key_case: snake
          key_case: camel
          # key_map_case: camel
          # param_case: pascal
          param_case: camel
          # string_interpolation: double_braces
          string_interpolation: braces
          # TODO: change to false
          flat_map: true
          # TODO: comment
          translation_overrides: true
          timestamp: true
          maps:
            - models.dashboardPeriod.name
            - models.dashboardPeriod.versus
            - models.currency.fullName
            - models.lengthUnit.name
            - models.weightUnit.name
            - models.orderStatus.name
            - models.paymentStatus.name
            - models.productStatus.name
            - models.paymentMethodType.name
            - models.paymentMethodType.description
            - models.storeStatus.name
            - models.promoType.name
            - models.businessVertical.name
            - models.salesChannel.name
            - models.contactType.name
            - models.onboardingStep.name
            - models.conversationType.name
            - models.adsProductStatus.name
            - exceptions.configUpdate
            - screens.main.nav
            - models.messengerType.name
            - models.conversationType.name
            - models.messageType.name
            - models.whatsAppAccountStatus.name
            - widgets.adsPreview.name
            - general.countries.name
            - models.deleteAccountReason.name
            - models.whatsappAccessRequestStatus.name
            - general.nationalities.name
            - models.adsStatus.name
            - models.adsGender.name
          pluralization:
            auto: cardinal
            cardinal:
              - someKey.apple
            ordinal:
              - someKey.place
          contexts:
            gender_context:
              enum:
                - male
                - female
              auto: false
              paths:
                - my.path.to.greet
          interfaces:
            PageData: onboarding.pages.*
            PageData2:
              paths:
                - my.path
                - cool.pages.*
              attributes:
                - String title
                - String? content
                ```

**Comment by guvanch07** [comment]
but I see backslash \ in the UI

**Comment by guvanch07** [comment]
yes I can handle it with replaseAll()
but it is not best solution

**Comment by Tienisto** [comment]
Maybe it's because you use the overriden version and not the generated string?
Because `String s = 'Hell\'o';` will not display the backslash `\`. You can try it out yourself.

**Comment by Tienisto** [comment]
As seen in the provided screenshot, the backslash and the apostrophe are shown in orange, hinting that the backslash is escaping the apostrophe. The only reason why the `\` is visible in the UI is that it is not correctly escaped when parsed from the "Translation Overrides" feature.

![Bildschirmfoto 2023-12-07 um 17 21 22](https://github.com/slang-i18n/slang/assets/38380847/76609bb8-0eac-4a72-ab4d-604673f3ac5a)

**Comment by guvanch07** [comment]
it is obvious))
but need to handle it

**Comment by guvanch07** [comment]
you can handle it with double quotes
`"Let's"` - like that

**Comment by Tienisto** [comment]
Fixed in `v3.26.2`
